server/src/main/java/org/elasticsearch/cluster/ClusterStatePublicationEvent.java

```diff
@@ -14,14 +14,30 @@
  */
 public class ClusterStatePublicationEvent {
 
+    private static final long NOT_SET = -1L;
```

Comment:
Out of curiosity, why not just use 0 for `NOT_SET` instead of having a separate flag for it, especially since when retrieving the stats we adjust NOT_SET back to 0 anyway?

Reply:
Really just to support the assertion that these things are set once and only once on each event. I added a comment in bd8ef0c6846.